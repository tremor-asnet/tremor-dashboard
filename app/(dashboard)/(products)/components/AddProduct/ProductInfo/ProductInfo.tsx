"use client";

// Libs
import dynamic from "next/dynamic";
import { Control, Controller, FieldErrors } from "react-hook-form";
import { KeyboardEvent } from "react";

// Components
import { Text } from "@tremor/react";
import InputField from "@/components/common/CustomField/InputField/InputField";
import SelectField from "@/components/common/CustomField/SelectField/SelectField";

// Constants
import { CATEGORY_PRODUCT, MESSAGES_ERROR } from "@/constants";
import { EXCEPT_KEYS } from "@/constants/common";

// Types
import { NewInfo } from "@/types";

// Css
import "@/styles/form.css";
import "react-quill/dist/quill.snow.css";
import "@/styles/quill.css";

const QuillEditor = dynamic(() => import("react-quill"), {
  ssr: false,
  loading: () => (
    <div className="flex justify-center items-center h-44 w-full">
      <p className="text-sm font-semibold text-[#7b809a] dark:text-white">
        Loading Quill Editor ...
      </p>
    </div>
  ),
});

interface ProductInfoProps {
  control: Control<NewInfo>;
  errors: FieldErrors<NewInfo>;
}

const ProductInfo = ({ control, errors }: ProductInfoProps) => {
  const errorNameMsg = errors.productName?.message;

  const handleWeightKeyDown = (e: KeyboardEvent<HTMLInputElement>) => {
    EXCEPT_KEYS.POSITIVE_DOUBLE.includes(e.key) && e.preventDefault();
  };

  const handleQuantityKeyDown = (e: KeyboardEvent<HTMLInputElement>) => {
    ["e", "E", "+", "-", "."].includes(e.key) && e.preventDefault();
  };

  return (
    <div className="grid grid-cols-1 sm:grid-cols-2 gap-6 items-start">
      <Controller
        name="productName"
        control={control}
        rules={{
          required: MESSAGES_ERROR.FIELD_REQUIRED,
          minLength: {
            value: 4,
            message: MESSAGES_ERROR.MIN_LENGTH_4,
          },
        }}
        render={({ field }) => (
          <div>
            <InputField
              type="text"
              isLabelTransform={true}
              label="Name"
              {...field}
            />
            <p className="product-info-err-msg">{errorNameMsg}</p>
          </div>
        )}
      />

      {/* Prevent input e, E, +, - to weight field */}
      <Controller
        name="weight"
        control={control}
        render={({ field }) => (
          <InputField
            type="number"
            label="Weight"
            isLabelTransform={true}
            {...field}
            onKeyDown={handleWeightKeyDown}
          />
        )}
      />

      <Controller
        name="description"
        control={control}
        render={({ field }) => (
          <div className="w-full flex flex-col h-48">
            <Text className="text-secondary dark:text-lighter mb-2">
              Description <span className="text-xs">(optional)</span>
            </Text>
            <QuillEditor
              theme="snow"
              placeholder="Some initial bold text"
              className="mt-2 dark:text-white"
              {...field}
            />
          </div>
        )}
      />

      <div className="flex flex-col gap-6">
        <Controller
          name="category"
          control={control}
          render={({ field }) => (
            <div>
              <h3 className="text-tremor-default text-secondary dark:text-white">
                Category
              </h3>
              <SelectField options={CATEGORY_PRODUCT} {...field} />
            </div>
          )}
        />

        <Controller
          name="quantity"
          control={control}
          render={({ field }) => (
            <InputField
              type="number"
              isLabelTransform={true}
              label="Quantity"
              onKeyDown={handleQuantityKeyDown}
              {...field}
            />
          )}
        />

        <Controller
          name="providerName"
          control={control}
          render={({ field }) => (
            <InputField
              type="text"
              label="Provider Name"
              isLabelTransform={true}
              {...field}
            />
          )}
        />
      </div>
    </div>
  );
};

export default ProductInfo;
