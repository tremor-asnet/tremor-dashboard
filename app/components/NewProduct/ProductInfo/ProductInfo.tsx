"use client";

import dynamic from "next/dynamic";

// Libs
import { Controller, useFormContext } from "react-hook-form";

// Components
import {
  Text,
  Flex,
  Card,
  TextInput,
  Select,
  SelectItem,
  NumberInput,
} from "@tremor/react";
const QuillEditor = dynamic(() => import("react-quill"), { ssr: false });

// Constants
import {
  MESSAGES_ERROR,
  CATEGORY_PRODUCT,
  NUMBER_REGEX,
  DECIMAL_REGEX,
} from "@/constants";

// Types
import { SelectOptionData } from "@/types";

// Styles
import "@/styles/form.css";
import "react-quill/dist/quill.snow.css"; // Import Quill styles

const ProductInfo = () => {
  const {
    control,
    formState: { errors },
  } = useFormContext();
  const { name } = errors || {};
  const nameErrorMessage = name?.message?.toString() || "";

  return (
    <Card className="w-full dark:bg-dark-tremor-primary rounded-lg shadow-box-icon-default ring-0">
      <Text className="text-primary dark:text-white font-bold text-xl mb-8">
        Product Information
      </Text>

      <Flex className="items-start gap-5">
        <Flex flexDirection="col" className="gap-4">
          <Controller
            control={control}
            rules={{
              required: MESSAGES_ERROR.NAME_REQUIRED,
              minLength: {
                value: 4,
                message: MESSAGES_ERROR.NAME_MIN_LENGTH,
              },
            }}
            render={({ field }) => (
              <div className="w-full">
                <Text className="text-secondary dark:text-lighter mb-2">
                  Name
                </Text>
                <TextInput
                  id="name"
                  placeholder="Name"
                  className="py-1 w-full dark:text-white hover:bg-transparent bg-transparent dark:bg-transparent focus:bg-transparent rounded-b-none border-l-0 border-r-0 border-t-0 border-b-1 focus:border-b-2 focus:outline-none focus:border-tremor-brand-subtle dark:border-light dark:focus:border-white shadow-none hover:bg-transparent ring-0"
                  autoFocus
                  required
                  {...field}
                />
                {nameErrorMessage && (
                  <p className="pt-1 text-[11px] xs:text-xs text-red-500">
                    {nameErrorMessage}
                  </p>
                )}
              </div>
            )}
            name="productName"
          />

          <Controller
            control={control}
            render={({ field }) => (
              <div className="w-full">
                <Text className="text-secondary dark:text-lighter mb-2">
                  Description <span className="text-xs">(optional)</span>
                </Text>
                <QuillEditor
                  placeholder="Content goes here..."
                  className="w-full rounded-md text-secondary dark:text-white"
                  {...field}
                />
              </div>
            )}
            name="description"
          />
        </Flex>
        <Flex flexDirection="col">
          <Controller
            control={control}
            rules={{
              pattern: { value: DECIMAL_REGEX, message: "Invalid weight" },
            }}
            render={({ field }) => (
              <div className="w-full mb-4">
                <Text className="text-secondary dark:text-lighter mb-2">
                  Weight
                </Text>
                <TextInput
                  id="weight"
                  placeholder="Weight"
                  className="py-1 w-full dark:text-white hover:bg-transparent bg-transparent dark:bg-transparent focus:bg-transparent rounded-b-none border-l-0 border-r-0 border-t-0 border-b-1 focus:border-b-2 focus:outline-none focus:border-tremor-brand-subtle dark:border-light dark:focus:border-white shadow-none hover:bg-transparent ring-0"
                  {...field}
                />
              </div>
            )}
            name="weight"
          />

          <Controller
            control={control}
            rules={{
              pattern: {
                value: NUMBER_REGEX,
                message: "Invalid quantity number",
              },
            }}
            render={({ field: { onChange, value } }) => (
              <div className="w-full mb-4">
                <Text className="text-secondary dark:text-lighter mb-2">
                  Quantity
                </Text>
                <NumberInput
                  enableStepper={false}
                  onValueChange={onChange}
                  value={value}
                  className="py-1 w-full dark:text-white hover:bg-transparent bg-transparent dark:bg-transparent focus:bg-transparent rounded-b-none border-l-0 border-r-0 border-t-0 border-b-1 focus:border-b-2 focus:outline-none focus:border-tremor-brand-subtle dark:border-light dark:focus:border-white shadow-none hover:bg-transparent ring-0"
                />
              </div>
            )}
            name="quantity"
          />

          <Controller
            control={control}
            render={({ field: { value, onChange } }) => (
              <div className="w-full mb-4">
                <Text className="text-secondary mb-3 dark:text-lighter mb-2">
                  Category
                </Text>
                <Select
                  placeholder="Clothing"
                  className="select-custom dark:text-white dark:border-light dark:focus:border-white"
                  value={value.toString()}
                  onValueChange={onChange}>
                  {CATEGORY_PRODUCT.map((item: SelectOptionData) => (
                    <SelectItem key={item.value} value={item.value}>
                      {item.option}
                    </SelectItem>
                  ))}
                </Select>
              </div>
            )}
            name="category"
          />
        </Flex>
      </Flex>
    </Card>
  );
};

export default ProductInfo;
