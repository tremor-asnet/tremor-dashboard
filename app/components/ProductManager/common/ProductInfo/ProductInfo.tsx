"use client";

// Libs
import dynamic from "next/dynamic";
import { Control, Controller, FieldErrors } from "react-hook-form";
import { KeyboardEvent } from "react";

// Components
import InputField from "@/components/common/CustomField/InputField/InputField";
import SelectField from "@/components/common/CustomField/SelectField/SelectField";

// Constants
import { CATEGORY_PRODUCT, MESSAGES_ERROR } from "@/constants";

// Types
import { NewInfo } from "@/types";

const QuillEditor = dynamic(() => import("react-quill"), { ssr: false });

interface ProductInfoProps {
  control: Control<NewInfo>;
  errors: FieldErrors<NewInfo>;
}

const ProductInfo = ({ control, errors }: ProductInfoProps) => {
  const errorNameMsg = errors.productName?.message;

  const handleWeightKeyDown = (e: KeyboardEvent<HTMLInputElement>) => {
    ["e", "E", "+", "-"].includes(e.key) && e.preventDefault();
  };

  const handleQuantityKeyDown = (e: KeyboardEvent<HTMLInputElement>) => {
    ["e", "E", "+", "-", "."].includes(e.key) && e.preventDefault();
  };

  return (
    <div className="grid grid-cols-1 sm:grid-cols-2 gap-6">
      <Controller
        name="productName"
        control={control}
        rules={{
          required: MESSAGES_ERROR.FIELD_REQUIRED,
          minLength: {
            value: 4,
            message: MESSAGES_ERROR.MIN_LENGTH_4,
          },
        }}
        render={({ field }) => (
          <div>
            <InputField type="text" label="Name" {...field} />
            <p className="product-info-err-msg">{errorNameMsg}</p>
          </div>
        )}
      />

      {/* Prevent input e, E, +, - to weight field */}
      <Controller
        name="weight"
        control={control}
        render={({ field }) => (
          <div>
            <InputField
              type="number"
              label="Weight"
              {...field}
              onKeyDown={handleWeightKeyDown}
            />
          </div>
        )}
      />

      <Controller
        name="description"
        control={control}
        render={({ field }) => (
          <div className="w-full">
            <h3 className="text-sm text-[#7b809a] dark:text-white">
              Description{" "}
              <span className="text-xs">&#x276A;optional&#x276B;</span>
            </h3>
            <QuillEditor
              placeholder="Content goes here..."
              className="mt-1 rounded-md text-secondary dark:text-white"
              {...field}
            />
          </div>
        )}
      />

      <div className="flex flex-col gap-6">
        <Controller
          name="category"
          control={control}
          render={({ field }) => (
            <div>
              <h3 className="text-xs text-[#7b809a] dark:text-white">
                Category
              </h3>
              <SelectField options={CATEGORY_PRODUCT} {...field} />
            </div>
          )}
        />

        <Controller
          name="quantity"
          control={control}
          render={({ field }) => (
            <div>
              <InputField
                type="number"
                label="Quantity"
                onKeyDown={handleQuantityKeyDown}
                {...field}
              />
            </div>
          )}
        />

        <Controller
          name="providerName"
          control={control}
          render={({ field }) => (
            <div>
              <InputField type="text" label="Provider Name" {...field} />
            </div>
          )}
        />
      </div>
    </div>
  );
};

export default ProductInfo;
