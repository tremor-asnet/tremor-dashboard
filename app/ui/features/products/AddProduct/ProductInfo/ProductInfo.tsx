"use client";

// Libs
import dynamic from "next/dynamic";
import { Control, Controller, FieldErrors } from "react-hook-form";
import { KeyboardEvent } from "react";

// Components
import { Text } from "@tremor/react";
import { InputField, SelectField } from "@/ui/components";

// Constants
import { CATEGORY_PRODUCT, MESSAGES_ERROR, EXCEPT_KEYS } from "@/constants";

// Types
import { NewInfo } from "@/types";

// Css
import "@/styles/form.css";
import "react-quill/dist/quill.snow.css";
import "@/styles/quill.css";

const QuillEditor = dynamic(() => import("react-quill"), {
  ssr: false,
  loading: () => (
    <div className="flex justify-center items-center h-44 w-full">
      <Text className="font-semibold text-secondary dark:text-white">
        Loading Quill Editor ...
      </Text>
    </div>
  ),
});

interface ProductInfoProps {
  control: Control<NewInfo>;
  errors: FieldErrors<NewInfo>;
}

const ProductInfo = ({ control, errors }: ProductInfoProps) => {
  const errorNameMsg = errors.productName?.message;

  const handleWeightKeyDown = (e: KeyboardEvent<HTMLInputElement>) => {
    EXCEPT_KEYS.POSITIVE_DOUBLE.includes(e.key) && e.preventDefault();
  };

  const handleQuantityKeyDown = (e: KeyboardEvent<HTMLInputElement>) => {
    ["e", "E", "+", "-", "."].includes(e.key) && e.preventDefault();
  };

  return (
    <div className="grid grid-cols-1 sm:grid-cols-2 gap-6 items-start">
      <Controller
        name="productName"
        control={control}
        rules={{
          required: MESSAGES_ERROR.FIELD_REQUIRED,
          minLength: {
            value: 4,
            message: MESSAGES_ERROR.MIN_LENGTH_4,
          },
        }}
        render={({ field }) => (
          <div>
            <InputField id="add-product-name" label="Name" {...field} />
            <Text className="pt-1 text-xs text-red-500 dark:text-red-500">
              {errorNameMsg}
            </Text>
          </div>
        )}
      />

      {/* Prevent input e, E, +, - to weight field */}
      <Controller
        name="weight"
        control={control}
        render={({ field }) => (
          <InputField
            id="add-product-weight"
            type="number"
            label="Weight"
            {...field}
            onKeyDown={handleWeightKeyDown}
          />
        )}
      />

      <Controller
        name="description"
        control={control}
        render={({ field }) => (
          <div className="w-full">
            <Text className="text-secondary text-sm dark:text-lighter">
              Description <span className="text-xs">(optional)</span>
            </Text>
            <QuillEditor
              theme="snow"
              placeholder="Some initial bold text"
              className="mt-3 dark:text-white"
              {...field}
            />
          </div>
        )}
      />

      <div className="flex flex-col gap-6">
        <div className="mb-2">
          <Controller
            name="category"
            control={control}
            render={({ field: { value, onChange } }) => {
              const convertedValue = value.toString();
              return (
                <SelectField
                  label="Category"
                  options={CATEGORY_PRODUCT}
                  value={convertedValue}
                  onChange={onChange}
                />
              );
            }}
          />
        </div>

        <div className="mb-2">
          <Controller
            name="quantity"
            control={control}
            render={({ field }) => (
              <InputField
                id="add-product-quantity"
                type="number"
                label="Quantity"
                onKeyDown={handleQuantityKeyDown}
                {...field}
              />
            )}
          />
        </div>

        <div className="mb-2">
          <Controller
            name="providerName"
            control={control}
            render={({ field }) => (
              <InputField
                id="add-product-provider"
                label="Provider Name"
                {...field}
              />
            )}
          />
        </div>
      </div>
    </div>
  );
};

export default ProductInfo;
