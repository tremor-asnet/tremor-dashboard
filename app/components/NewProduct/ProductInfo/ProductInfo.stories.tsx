import type { Meta, StoryObj } from "@storybook/react";

// Components
import { Flex } from "@tremor/react";
import ProductInfo from "./ProductInfo";

const meta = {
  title: "Components/ProductInfo",
  component: ProductInfo,
  tags: ["autodocs"],
} as Meta<typeof ProductInfo>;

export default meta;

type Story = StoryObj<typeof meta>;

export const NewProductInfo: Story = {
  render: () => (
    <Flex className="w-full justify-center mt-10">
      <ProductInfo />
    </Flex>
  ),
};

export const EditProductInfo: Story = {
  render: () => (
    <Flex className="w-full justify-center mt-10">
      <ProductInfo />
    </Flex>
  ),
};
