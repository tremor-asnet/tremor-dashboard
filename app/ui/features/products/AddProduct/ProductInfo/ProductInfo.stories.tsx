import type { Meta, StoryObj } from "@storybook/react";

// Libs
import { useForm } from "react-hook-form";

// Components
import ProductInfo from "./ProductInfo";

// Types
import { NewInfo } from "@/types";

const meta = {
  title: "Components/Products/AddProduct/ProductInfo",
  component: ProductInfo,
  tags: ["autodocs"],
  argTypes: {
    control: {
      description:
        "This is control from useForm of react-hook-form with type NewInfo({productName: string, description: string, providerName: string, weight: string, category: string, quantity: string})",
    },
  },
} as Meta<typeof ProductInfo>;

export default meta;

type Story = StoryObj<typeof meta>;

const ProductInfoFormProvider = () => {
  const { control } = useForm<NewInfo>({
    defaultValues: {
      productName: "Product A",
      description: "",
      weight: 0,
      category: 1,
      quantity: 0,
      providerName: "Provider A",
    },
  });

  return <ProductInfo control={control} />;
};

export const Default: Story = {
  render: () => <ProductInfoFormProvider />,
};
