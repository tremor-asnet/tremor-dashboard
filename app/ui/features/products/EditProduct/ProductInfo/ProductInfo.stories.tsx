// Libs
import { ReactNode, FC } from "react";
import { FormProvider, useForm } from "react-hook-form";
import type { Meta, StoryObj } from "@storybook/react";

// Components
import ProductInfo from "./ProductInfo";

const StorybookFormProvider: FC<{ children: ReactNode }> = ({ children }) => {
  const methods = useForm();
  return <FormProvider {...methods}>{children}</FormProvider>;
};

const meta = {
  title: "Components/Products/EditProduct/ProductInfo",
  component: ProductInfo,
  tags: ["autodocs"],
} as Meta<typeof ProductInfo>;

export default meta;

type Story = StoryObj<typeof meta>;

export const Primary: Story = {
  render: () => (
    <StorybookFormProvider>
      <ProductInfo />
    </StorybookFormProvider>
  ),
};
