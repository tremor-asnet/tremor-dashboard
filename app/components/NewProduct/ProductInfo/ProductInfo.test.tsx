import { render } from "@testing-library/react";

// Components
import ProductInfo from "./ProductInfo";

describe("Testing ProductInfo component", () => {
  // TODO: Update unit test later
  it.skip("Should match snapshot", () => {
    const component = render(<ProductInfo />);
    expect(component).toMatchSnapshot();
  });
});
