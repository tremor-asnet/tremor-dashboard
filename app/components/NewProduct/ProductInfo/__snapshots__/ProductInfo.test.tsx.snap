// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ProductInfo component Should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="tremor-Card-root relative text-left rounded-tremor-default bg-tremor-background ring-tremor-ring shadow-tremor-card dark:ring-dark-tremor-ring dark:shadow-dark-tremor-card border-tremor-brand dark:border-dark-tremor-brand p-6 w-full dark:bg-dark-tremor-primary rounded-lg shadow-box-icon-default ring-0"
      >
        <p
          class="text-tremor-default text-primary dark:text-white font-bold text-xl mb-8"
        >
          Product Information
        </p>
        <form
          class="product-info"
        >
          <div
            class="tremor-Flex-root flex w-full flex-col justify-between items-start gap-6 md:flex-row"
          >
            <div
              class="tremor-Flex-root flex w-full flex-col justify-between items-center gap-6"
            >
              <div
                class="w-full mb-2"
              >
                <p
                  class="text-tremor-default text-secondary dark:text-lighter mb-2"
                >
                  Name
                </p>
                <div
                  class="tremor-TextInput-root relative flex items-center min-w-[10rem] outline-none rounded-tremor-default transition duration-100 shadow-tremor-input dark:shadow-dark-tremor-input dark:hover:bg-dark-tremor-background-muted text-tremor-content border-tremor-brand-subtle ring-tremor-brand-muted dark:ring-dark-tremor-brand-muted border py-1 w-full dark:text-white bg-transparent dark:bg-transparent focus:bg-transparent rounded-b-none border-l-0 border-r-0 border-t-0 border-b focus:outline-none focus:border-tremor-brand-subtle dark:border-light dark:focus:border-white shadow-none hover:bg-transparent ring-0"
                >
                  <input
                    class="tremor-TextInput-input w-full focus:outline-none focus:ring-0 border-none bg-transparent text-tremor-default rounded-tremor-default transition duration-100 text-tremor-content-emphasis dark:text-dark-tremor-content-emphasis [appearance:textfield] [&::-webkit-outer-spin-button]:appearance-none [&::-webkit-inner-spin-button]:appearance-none pl-3 pr-4 py-2 placeholder:text-tremor-content dark:placeholder:text-dark-tremor-content"
                    data-testid="base-input"
                    id="name"
                    placeholder="Name"
                    required=""
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div
                class="w-full"
              >
                <p
                  class="text-tremor-default text-secondary dark:text-lighter mb-2"
                >
                  Description 
                  <span
                    class="text-xs pl-1"
                  >
                    (optional)
                  </span>
                </p>
              </div>
            </div>
            <div
              class="tremor-Flex-root flex w-full flex-col justify-between items-start"
            >
              <div
                class="w-full mb-8"
              >
                <p
                  class="text-tremor-default text-secondary dark:text-lighter mb-2"
                >
                  Weight
                </p>
                <input
                  class="py-[12px] w-full text-tremor-default dark:text-white hover:bg-transparent bg-transparent dark:bg-transparent focus:bg-transparent rounded-b-none border-l-0 border-r-0 border-t-0 border-b focus:outline-none border-gray focus:border-tremor-brand-subtle dark:border-light dark:focus:border-white shadow-none hover:bg-transparent ring-0"
                  id="weight"
                  placeholder="Weight"
                  type="number"
                  value="0"
                />
              </div>
              <div
                class="w-full mb-8 sm:max-w-auto"
              >
                <p
                  class="text-tremor-default text-secondary dark:text-lighter mb-2"
                >
                  Quantity
                </p>
                <input
                  class="py-[12px] w-full text-tremor-default dark:text-white hover:bg-transparent bg-transparent dark:bg-transparent focus:bg-transparent rounded-b-none border-l-0 border-r-0 border-t-0 border-b focus:outline-none border-gray focus:border-tremor-brand-subtle dark:border-light dark:focus:border-white shadow-none hover:bg-transparent ring-0"
                  id="quantity"
                  placeholder="Quantity"
                  type="number"
                  value="0"
                />
              </div>
              <div
                class="w-full mb-7"
              >
                <div
                  class="tremor-Flex-root flex w-full justify-between flex-col items-start"
                >
                  <p
                    class="text-tremor-default text-secondary h-5 mb-3 dark:text-lighter"
                  >
                    Category
                  </p>
                  <div
                    class="min-w-[10rem] relative text-tremor-default py-1 select-custom w-full dark:text-white bg-transparent dark:bg-transparent focus:bg-transparent rounded-b-none border-l-0 border-r-0 border-t-0 border-gray hover:border-black active:border-black focus:outline-none focus:border-tremor-brand-subtle dark:border-light dark:focus:border-white shadow-none hover:bg-transparent ring-0"
                    data-headlessui-state=""
                    id="category"
                  >
                    <button
                      aria-expanded="false"
                      aria-haspopup="listbox"
                      class="w-full outline-none text-left whitespace-nowrap truncate rounded-tremor-default focus:ring-2 transition duration-100 shadow-tremor-input focus:border-tremor-brand-subtle focus:ring-tremor-brand-muted dark:shadow-dark-tremor-input dark:focus:border-dark-tremor-brand-subtle dark:focus:ring-dark-tremor-brand-muted pl-3 pr-8 py-2 border bg-tremor-background dark:bg-dark-tremor-background hover:bg-tremor-background-muted dark:hover:bg-dark-tremor-background-muted text-tremor-content dark:text-dark-tremor-content border-tremor-border dark:border-dark-tremor-border"
                      data-headlessui-state=""
                      id="headlessui-listbox-button-:r0:"
                      type="button"
                    >
                      <span
                        class="w-[90%] block truncate"
                      >
                        Clothing
                      </span>
                      <span
                        class="absolute inset-y-0 right-0 flex items-center mr-3"
                      >
                        <svg
                          class="tremor-Select-arrowDownIcon flex-none text-tremor-content-subtle dark:text-dark-tremor-content-subtle h-4 w-4"
                          fill="none"
                          stroke="currentColor"
                          stroke-width="2.5"
                          viewBox="0 0 24 24"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M19 9l-7 7-7-7"
                            stroke-linecap="round"
                            stroke-linejoin="round"
                          />
                        </svg>
                      </span>
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="tremor-Card-root relative text-left rounded-tremor-default bg-tremor-background ring-tremor-ring shadow-tremor-card dark:ring-dark-tremor-ring dark:shadow-dark-tremor-card border-tremor-brand dark:border-dark-tremor-brand p-6 w-full dark:bg-dark-tremor-primary rounded-lg shadow-box-icon-default ring-0"
    >
      <p
        class="text-tremor-default text-primary dark:text-white font-bold text-xl mb-8"
      >
        Product Information
      </p>
      <form
        class="product-info"
      >
        <div
          class="tremor-Flex-root flex w-full flex-col justify-between items-start gap-6 md:flex-row"
        >
          <div
            class="tremor-Flex-root flex w-full flex-col justify-between items-center gap-6"
          >
            <div
              class="w-full mb-2"
            >
              <p
                class="text-tremor-default text-secondary dark:text-lighter mb-2"
              >
                Name
              </p>
              <div
                class="tremor-TextInput-root relative flex items-center min-w-[10rem] outline-none rounded-tremor-default transition duration-100 shadow-tremor-input dark:shadow-dark-tremor-input dark:hover:bg-dark-tremor-background-muted text-tremor-content border-tremor-brand-subtle ring-tremor-brand-muted dark:ring-dark-tremor-brand-muted border py-1 w-full dark:text-white bg-transparent dark:bg-transparent focus:bg-transparent rounded-b-none border-l-0 border-r-0 border-t-0 border-b focus:outline-none focus:border-tremor-brand-subtle dark:border-light dark:focus:border-white shadow-none hover:bg-transparent ring-0"
              >
                <input
                  class="tremor-TextInput-input w-full focus:outline-none focus:ring-0 border-none bg-transparent text-tremor-default rounded-tremor-default transition duration-100 text-tremor-content-emphasis dark:text-dark-tremor-content-emphasis [appearance:textfield] [&::-webkit-outer-spin-button]:appearance-none [&::-webkit-inner-spin-button]:appearance-none pl-3 pr-4 py-2 placeholder:text-tremor-content dark:placeholder:text-dark-tremor-content"
                  data-testid="base-input"
                  id="name"
                  placeholder="Name"
                  required=""
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="w-full"
            >
              <p
                class="text-tremor-default text-secondary dark:text-lighter mb-2"
              >
                Description 
                <span
                  class="text-xs pl-1"
                >
                  (optional)
                </span>
              </p>
            </div>
          </div>
          <div
            class="tremor-Flex-root flex w-full flex-col justify-between items-start"
          >
            <div
              class="w-full mb-8"
            >
              <p
                class="text-tremor-default text-secondary dark:text-lighter mb-2"
              >
                Weight
              </p>
              <input
                class="py-[12px] w-full text-tremor-default dark:text-white hover:bg-transparent bg-transparent dark:bg-transparent focus:bg-transparent rounded-b-none border-l-0 border-r-0 border-t-0 border-b focus:outline-none border-gray focus:border-tremor-brand-subtle dark:border-light dark:focus:border-white shadow-none hover:bg-transparent ring-0"
                id="weight"
                placeholder="Weight"
                type="number"
                value="0"
              />
            </div>
            <div
              class="w-full mb-8 sm:max-w-auto"
            >
              <p
                class="text-tremor-default text-secondary dark:text-lighter mb-2"
              >
                Quantity
              </p>
              <input
                class="py-[12px] w-full text-tremor-default dark:text-white hover:bg-transparent bg-transparent dark:bg-transparent focus:bg-transparent rounded-b-none border-l-0 border-r-0 border-t-0 border-b focus:outline-none border-gray focus:border-tremor-brand-subtle dark:border-light dark:focus:border-white shadow-none hover:bg-transparent ring-0"
                id="quantity"
                placeholder="Quantity"
                type="number"
                value="0"
              />
            </div>
            <div
              class="w-full mb-7"
            >
              <div
                class="tremor-Flex-root flex w-full justify-between flex-col items-start"
              >
                <p
                  class="text-tremor-default text-secondary h-5 mb-3 dark:text-lighter"
                >
                  Category
                </p>
                <div
                  class="min-w-[10rem] relative text-tremor-default py-1 select-custom w-full dark:text-white bg-transparent dark:bg-transparent focus:bg-transparent rounded-b-none border-l-0 border-r-0 border-t-0 border-gray hover:border-black active:border-black focus:outline-none focus:border-tremor-brand-subtle dark:border-light dark:focus:border-white shadow-none hover:bg-transparent ring-0"
                  data-headlessui-state=""
                  id="category"
                >
                  <button
                    aria-expanded="false"
                    aria-haspopup="listbox"
                    class="w-full outline-none text-left whitespace-nowrap truncate rounded-tremor-default focus:ring-2 transition duration-100 shadow-tremor-input focus:border-tremor-brand-subtle focus:ring-tremor-brand-muted dark:shadow-dark-tremor-input dark:focus:border-dark-tremor-brand-subtle dark:focus:ring-dark-tremor-brand-muted pl-3 pr-8 py-2 border bg-tremor-background dark:bg-dark-tremor-background hover:bg-tremor-background-muted dark:hover:bg-dark-tremor-background-muted text-tremor-content dark:text-dark-tremor-content border-tremor-border dark:border-dark-tremor-border"
                    data-headlessui-state=""
                    id="headlessui-listbox-button-:r0:"
                    type="button"
                  >
                    <span
                      class="w-[90%] block truncate"
                    >
                      Clothing
                    </span>
                    <span
                      class="absolute inset-y-0 right-0 flex items-center mr-3"
                    >
                      <svg
                        class="tremor-Select-arrowDownIcon flex-none text-tremor-content-subtle dark:text-dark-tremor-content-subtle h-4 w-4"
                        fill="none"
                        stroke="currentColor"
                        stroke-width="2.5"
                        viewBox="0 0 24 24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M19 9l-7 7-7-7"
                          stroke-linecap="round"
                          stroke-linejoin="round"
                        />
                      </svg>
                    </span>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
